ui/src/views/Tasks/TaskIndex/IndexedTask/index.jsx

```diff
@@ -90,6 +91,21 @@ export default class IndexedTask extends Component {
     });
   };
 
+  handleTaskNamespaceSearchSubmit = taskNamespaceSearch => {
+    const searchText = taskNamespaceSearch.replace(/\s+/g, '');
+    const namespace = searchText
+      .split('.')
+      .slice(0, -1)
+      .join('.');
+    const taskFromNamespace = searchText.split('.').slice(-1)[0];
+
+    if (namespace && taskFromNamespace) {
```

Comment:
@helfi92 Is this fine?

Here are the possible ways someone might interact with the search bar and the obtained results.

**No task in the namespace**

If a user just searches for 'gecko', nothing would happen as there is no task identified from the namespace.

**Namespace and task present**

If a user searches for `ci.cache.level-3.docker-images.v2.hg-push.latest`, they will be taken to `/tasks/index/ci.cache.level-3.docker-images.v2.hg-push/latest` which will render the `IndexedTask` component.

But if search for `ci.cache.level-3.docker-images.v2.hg-push`, they will be taken to `/tasks/index/ci.cache.level-3.docker-images.v2/hg-push` which won't work and an error will be rendered.

Should we worry about this? If yes, how do we handle this scenario?

Reply:
No worries!